sources/packages/api/src/route-controllers/student/student.controller.ts

```diff
@@ -308,4 +313,48 @@ export class StudentController extends BaseController {
 
     stream.pipe(response);
   }
+
+  @Get("application-summary")
+  async getStudentApplicationSummary(
+    @UserToken() userToken: IUserToken,
+  ): Promise<StudentApplicationDTO[]> {
+    const existingStudent = await this.studentService.getStudentByUserName(
+      userToken.userName,
+    );
+    if (!existingStudent) {
+      throw new NotFoundException(
+        `No student was found with the student name ${userToken.userName}`,
+      );
+    }
+    const application = await this.applicationService.getAllStudentApplications(
+      existingStudent.id,
+    );
+    let response = [] as StudentApplicationDTO[];
+    for await (const element of application) {
+      let offering = undefined;
+      if (
+        element.data?.selectedProgram &&
+        element.data?.offeringIWillBeAttending
```

Comment:
The offering that students will be attending should come from the offering_id on the application table. This is the field that will be resolved by the PIR or populated by the workflow if it is present in application data.
@abschwenker, is there any concern about it? The only drawback is that the offering_id could be null till it is populated by the workflow or resolved by the PIR, but I believe that offering_id should be the source of thuth.

Reply:
Null does seem logical until he PIR is complete because the Offering ID will contain the final source of truth